src/sync-pending-events.ts

```diff
@@ -101,28 +121,46 @@ export class EventTracker {
     this.contract = web3.getContract(dataJson, this.address, monitorAccount);
     this.dbUrl = mongodbUrl;
     this.dbName = networkId + this.address;
+    this.source = source;
+    const defaultStep = 2000;
+    if(eventsSyncStep == undefined || eventsSyncStep <= 0){
+      this.eventsSyncStep = defaultStep;
+    }else{
+      this.eventsSyncStep = eventsSyncStep;
+    }
   }
 
   private async syncPastEvents(
     handlers: (n: string, v: any, hash: string) => Promise<void>,
     db: EventDBHelper
   ) {
-    let lastblock = await db.getLastMonitorBlock();
-    console.log("sync from ", lastblock);
+    let checkFromBlockNumber;
+    let lastUpdatedBlock = await db.getLastMonitorBlock();
+    let zeroEventCheckedBlockNumber = await db.getZeroEventCheckedBlockNumber();
+    let latestBlockNumber = await getLatestBlockNumber(this.source);
+    if(lastUpdatedBlock == 0){
+      checkFromBlockNumber = zeroEventCheckedBlockNumber;
+      await db.updateZeroEventCheckedBlockNumber(latestBlockNumber);
+    }else{
+      checkFromBlockNumber = lastUpdatedBlock;
+    }
+    console.log("sync from ", checkFromBlockNumber);
     try {
-      let pastEvents = await this.contract.getPastEventsFrom(lastblock + 1);
-      console.log("sync from ", lastblock, "done");
-      for (let r of pastEvents) {
-        console.log(
-          "========================= Get L1 Event: %s ========================",
-          r.event
-        );
-        console.log("blockNumber:", r.blockNumber);
-        console.log("blockHash:", r.blockHash);
-        console.log("transactionHash:", r.transactionHash);
-        let e = buildEventValue(this.l1Events, r);
-        await handlers(r.event, e, r.transactionHash);
-        await db.updateLastMonitorBlock(r, e);
+      let pastEvents = await this.contract.getPastEventsFromSteped(checkFromBlockNumber + 1, latestBlockNumber, this.eventsSyncStep);
+      console.log("sync from ", checkFromBlockNumber, "done");
+      for(let group of pastEvents){
+        for (let r of group) {
+          console.log(
+            "========================= Get L1 Event: %s ========================",
+            r.event
+          );
+          console.log("blockNumber:", r.blockNumber);
+          console.log("blockHash:", r.blockHash);
+          console.log("transactionHash:", r.transactionHash);
+          let e = buildEventValue(this.l1Events, r);
+          await handlers(r.event, e, r.transactionHash);
+          await db.updateLastMonitorBlock(r, e);
+        }
```

Comment:
update lastestBlockNumber if this round finish successfully, so that if no events are happen within this round, we can start from a better start point next time.